src/index_runner/es_indexers/annotated_metagenome_assembly.py

```diff
@@ -0,0 +1,131 @@
+# KBaseMetagenomes.AnnotatedMetagenomeAssembly indexer
+from src.index_runner.es_indexers.indexer_utils import mean, handle_id_to_file
+
+import json
+import gzip
+import shutil
+import os
+
+_NAMESPACE = "WS"
+_AMA_INDEX_VERSION = 1
+_AMA_FEATURES_INDEX_VERSION = 1
+_AMA_INDEX_NAME = "annotated_metagenome_assembly:" + str(_AMA_INDEX_VERSION)
+_AMA_FEATURES_INDEX_NAME = "annotated_metagenome_assembly_features:" + str(_AMA_FEATURES_INDEX_VERSION)
+
+_DIR = os.path.dirname(os.path.realpath(__file__))
+
+
+def _index_ama(features_file_gz_path, data, ama_id):
+    """"""
+    publication_titles = [pub[2] for pub in data.get('publications', [])]
+    publication_authors = [pub[5] for pub in data.get('publications', [])]
+    ama_index = {
+        '_action': 'index',
+        'doc': {
+            'size': data.get('dna_size'),
+            'source_id': data.get('source_id'),
+            'source': data.get('source'),
+            'gc_content': data.get('gc_content'),
+            'warnings': data.get('warnings'),
+            'num_contigs': data.get('num_contigs'),
+            'mean_contig_length': mean(data.get('contig_lengths', [])),
+            'external_source_origination_date': data.get('external_source_origination_date'),
+            'original_source_file_name': data.get('original_source_file_name'),
+            'environment': data.get('environment'),
+            'num_features': data.get('num_features'),
+            'publication_authors': publication_authors,
+            'publication_titles': publication_titles,
+            'molecule_type': data.get('molecule_type'),
+            'assembly_ref': data.get('assembly_ref'),
+            'notes': data.get('notes'),
+            # not sure what to do with the following fields.
+            # list<Ontology_event> ontology_events;
+            # mapping<string, mapping<string, string>> ontologies_present;
+        },
+        'index': _AMA_INDEX_NAME,
+        'id': ama_id
+    }
+    yield ama_index
+
+    # unzip gzip file.
+    features_file_path = _DIR + "/features.json"
+    with gzip.open(features_file_gz_path, "rb") as f_in:
+        with open(features_file_path, "wb") as f_out:
+            shutil.copyfileobj(f_in, f_out)
+
+    with open(features_file_path) as f:
+        features = json.load(f)
+
+    for feat in features:
+        id_ = feat.get('id')
+        feat_id = ama_id + f"::ama_ft::{id_}"
+        if feat.get('dna_sequence'):
+            dna_seq = feat.get('dna_sequence')
+            feat_gc_content = ((float(dna_seq.lower().count('c')) + float(dna_seq.lower().count('g'))) / len(dna_seq))
```

Comment:
Is this really not in the source object and is needed for search? Seems a little weird to calculate it here